File: LinkedList.js
class Node {
  constructor(value) {
    this.value = value
    this.next = null
  }
}

class LinkedList {
  constructor() {
    this.head = null
    this.tail = null
  }

  AddToBack(value) {
    let ref = new Node(value)
    if (this.head === null) {
      this.head = ref
      this.tail = ref
    } else {
      this.tail.next = ref
      this.tail = ref
    }
  }

  AddToFront(value) {
    let ref = new Node(value)
    if (this.head === null) {
      this.head = ref
      this.tail = ref
    } else {
      ref.next = this.head;
      this.head = ref
    }
  }

  AddAfter(searchValue, newValue) {
    for (let current = this.head; current != null; current = current.next) {
      if (current.value === searchValue) {
        if (current === this.tail) {
          this.Add(newValue)
          return true
        }
        let ref = new Node(newValue)
        ref.next = current.next
        current.next = ref
        return true
      }
    }
  }

  Print() {
    for (let current = this.head; current != null; current = current.next) {
      console.log(current.value)
    }
  }
}
console.clear()

let l = new LinkedList()
l.AddToBack(1)
l.AddToBack(2)
l.AddToBack(3)
l.AddToFront(4)
//l.Print()
l.AddAfter(4, 4)
l.Print()

File: LinkedList.cs
using System;
namespace Data_Structures
{
    public class LinkedList
    {
        public SglNode head;
        public SglNode tail;

        public LinkedList()
        {
            this.head = this.tail = null;
        }

        public void Add(int value){
            SglNode newNode = new SglNode(value);
            if(head==null) head = tail = newNode;
            else tail.next=newNode;
            tail = newNode;
        }

        public void AddToBack(int value){
            SglNode newNode = new SglNode(value);
            if(tail==null){
                head = tail = newNode;
            }
            else{
                tail.next = newNode;
                tail = newNode;
            }
        }

        public void AddToFront(int value){
            SglNode newNode = new SglNode(value);
            if(head==null) {
                head = tail = newNode;
            }
            else {
                newNode.next = head;
                head = newNode;
            }
        }

        public void Delete(int value){
            if(head.value.ToString() == value.ToString()){
                head = head.next;
                return;
            }
            for (SglNode current = head; current!=null;  current = current.next)
            {
                if(current.next.value.ToString() == value.ToString()){
                    if(current.next==tail){
                        tail=current;
                        current.next=null;
                    }
                    else
                        current.next = current.next.next;
                    break;
                }

            }
        }

        public override string ToString(){
            string listStr = "";
            for (SglNode current = head; current!=null;  current = current.next)
            {
                if(listStr.Length!=0) listStr+=" - ";
                listStr+=$"{current.value}";
            }
            return listStr;
        }
    }
}